api/v1beta1/types_class.go

```diff
@@ -160,3 +160,8 @@ func (sgc *SecurityGroupClass) setDefaults() {
 		}
 	}
 }
+
+// setDefaults sets default values for NatGatewayClass.
+func (natgaw *NatGatewayClassSpec) setDefaults(clusterTemplateName string, nodeSubnetCounter int) {
+	natgaw.Name = withIndex(generateNatGatewayName(clusterTemplateName), nodeSubnetCounter)
```

Comment:
`clusterTemplateName` is still not unique enough since it will be shared between clusters using the same template. That being said, I don't think we need to do the defaulting here in the ClusterClass spec. It seems that it's similar to Subnet Name in that Subnet.Name is a required field but it doesn't get defaulted in the template itself, it only gets defaulted per cluster in `azurecluster_default.go` using the cluster name as input to generate a subnet name. @willie-yao does this make sense to you?

Reply:
Yes that makes sense. The name shouldn't be shared across clusters in the ClusterClass. The defaulting here shouldn't be needed as it is happening in [azurecluster_default](https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/3365/files#diff-016ca697ac4323c794d5dda09a16fd6152518acb4e3da6efef765fe5011a796bR132).

Reply:
Thanks @CecileRobertMichon and @willie-yao ! Now, I have reverted default setting of NAT gateway for `AzureClusterTemplate` object. And during validation phase, use `IsIPv6Enabled` instead of `IsNatGatewayEnabled` to judge if we need the outboundLB or not. Please help review again, thanks again!